src/main/java/bio/terra/cli/command/resource/CheckAccess.java

```diff
@@ -25,6 +26,9 @@ public class CheckAccess extends WsmBaseCommand {
   protected void execute() {
     workspaceOption.overrideIfSpecified();
     Resource resource = Context.requireWorkspace().getResource(resourceNameOption.name);
+    if (resource.getResourceType().equals(Resource.Type.DATAPROC_CLUSTER)) {
```

Comment:
Do we need this check here? The resource is visible in display / list resources. 
check access might only read sam. 

overall behavior is a little inconsistent wrt flags. 
Are the flags intended to be removed soon? if so, please add a ticket to track it. 
If the flag check is meant to be retained for medium - long term, this needs to be updated. 

overall --
1) if data proc is not supported,. should list resources display them? 
2) is flag=false, operations are not allowed. Some like check-access may not need this feature-flagcheck. However the same operation may be allowed via Workspace operation (example delete)

I would be open to handling it in a separate PR if the above needs discussion / complex changes

Reply:
One option - check support only on create, not on read. This is often used for aws